src/test/java/uk/gov/justice/probation/courtcaseservice/service/OffenderServiceTest.java

```diff
@@ -145,4 +159,46 @@ public void whenGetConvictionRequirements_returnRequirements() {
         assertThat(requirements).isSameAs(expectedRequirements);
         verify(offenderRestClient).getConvictionRequirements(CRN, CONVICTION_ID);
     }
+
+    @DisplayName("Getting probation record does not throw exception when oasys assessment data is missing")
+    @Test
+    public void givenAssessmentNotFound_whenGetOffender_thenDoNotThrowException() {
+        when(offenderRestClient.getProbationRecordByCrn(CRN)).thenReturn(Mono.just(ProbationRecord.builder().crn(CRN).build()));
+        when(offenderRestClient.getConvictionsByCrn(CRN)).thenReturn(Mono.just(singletonList(conviction)));
+        when(offenderRestClient.getDocumentsByCrn(CRN)).thenReturn(Mono.just(groupedDocuments));
+        // throw OffenderNotFoundException to simulate a 404 returned by assessments api
+        when(assessmentsRestClient.getAssessmentByCrn(CRN)).thenReturn(Mono.error(new OffenderNotFoundException(CRN)));
+
+        ProbationRecord probationRecord = service.getProbationRecord(CRN, false);
+        assertThat(probationRecord).isNotNull();
+
+        // the assessment field should just be empty
+        assertThat(probationRecord.getAssessment()).isNull();
+
+        // but the rest of the record should be populated as normal
+        assertThat(probationRecord.getConvictions()).hasSize(1);
+        final Conviction conviction = probationRecord.getConvictions().get(0);
+        assertThat(conviction.getDocuments()).hasSize(2);
+    }
+
+    @DisplayName("Getting probation record does not throw exception when assessment api fails for any reason")
```

Comment:
Hmm, not sure about this. The danger is we a) suppress errors that we need to fix and b) give the user the impression that data doesn't exist in the system when in practice it just isn't available due to an error. 

The 'a' we can mitigate by logging an error I think 'b' is more dangerous, we don't want to give probation staff misleading information as that might ultimately affect what sentence is passed.